### DISCUSSION TITLE
[27 marzo 2023] Calcolabilità

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/automi-calcolabilita-complessita#8] [Calcolabilità]
![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/69389461/ae752348-0ee8-4b6a-bc50-9961604e2d5c)

**Comment by Exyss** [comment]
Per il primo punto, ripropongo la soluzione presente sui miei appunti:

![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/65043207/e6ffbe3a-561a-4f29-9fb6-147d9c183807)

----

La dimostrazione richiesta nel secondo punto è presente sul libro di testo

**Reply by 18xChris** [reply]
Per il secondo punto propongo la soluzione presente sugli appunti di [CasuFrost](https://github.com/CasuFrost/University_notes)
![image](https://github.com/user-attachments/assets/a7b3bb71-c4b9-43e2-b5bb-3aa6261c1c1a)

**Comment by cornflexxx** [comment]
Supponiamo $W_{TM}\in R$, con decisore $D$, allora possiamo costruire un decisore T  per $A_{TM}$:
1. Input $\langle M,w\rangle$
2. Costruisce la $TM$ $M'$: 

-  Su input $x$ : esegue $M(x)$, se $M$ accetta allora stampa $a$ e **accetta**
- altrimenti **rifiuta**

3. eseguiamo $D(\langle M',w,a\rangle)$ se accetta $\implies$ **$T$ accetta**

**Comment by daromacarlo** [comment]
<img width="1212" height="816" alt="immagine" src="https://github.com/user-attachments/assets/c6ceee40-ee37-468e-b728-2cc8a34dc547" />
<img width="1213" height="605" alt="immagine" src="https://github.com/user-attachments/assets/807cf98b-1a4b-4516-9c59-b70537f54839" />
